src/components/notes/editor.jsx

```diff
@@ -92,6 +98,16 @@ export default function Editor(props) {
     cancelLabel: t('Notes.Editor.exit_confirmation_cancel')
   })
 
+  const [isTrashed, setTrashed] = useState(false)
+
+  useEffect(() => {
+    setTrashed(!!doc?.file?.attributes?.trashed)
+  }, [doc])
+
+  const handleRestore = () => {
+    setTrashed(false)
+  }
```

Comment:
I don't like the idea of having to handle manually the trashed attribute. Specially since `useNote` is handling it. 

I think you've done that because the `useNote` do not update by itself after the call on `await client.collection('io.cozy.files').restore(noteId)`, right? 

It is not updated because calling the collection directly bypass the cozy-client redux store. 

So we have 3 choices: 
- Call a callback => The choice you have done, but now we have several source for the data and one can be outdated. => KO.
- Expose a client.XXX in order to not bypass the redux store of cozy-client. => Let's see if we can do something about it
- Call a dispatch manually after the call to `restore()`. Like what we do in the `RealTimeQuery` component. 
```
const updatedNote = await client.collection('io.cozy.files').restore(id)
client.dispatch(
    receiveMutationResult(
      client.generateRandomId(),
      updatedNote,
      {},
     Mutations.updateDocument(data)
    )
  )
  ```
  
By doing something like that, the `useNote` should be update to date

Reply:
Me too, but as you guesse, the document doesn't update automatically. I like your proposal, but useNote doesn't use the cozy-client redux store to retrieve information, it retrieves it directly from the stack client ([code](https://github.com/cozy/cozy-notes/blob/3b2fd1e6f109a3e262e6d6de179c039d546f0887/src/lib/collab/stack-client.js#L276)). I'll see if I can't do it some other way to get the updated information.